x-pack/plugin/esql/compute/gen/src/main/java/org/elasticsearch/compute/gen/AggregatorImplementer.java

```diff
@@ -84,21 +89,24 @@ public AggregatorImplementer(
         this.declarationType = declarationType;
         this.warnExceptions = warnExceptions;
 
-        this.init = findRequiredMethod(declarationType, new String[] { "init", "initSingle" }, e -> true);
+        this.init = requireStaticMethod(
+            declarationType,
+            // This should be more restrictive and require org.elasticsearch.compute.aggregation.AggregatorState
```

Comment:
It is enforced in https://github.com/elastic/elasticsearch/pull/121898/commits/57d34e4ba03f2e315ff1a851ad6218a181c98bf7 please let me know if that should be merged as well.